application/src/main/java/org/opentripplanner/graph_builder/module/osm/VertexGenerator.java

```diff
@@ -95,24 +123,51 @@ IntersectionVertex getVertexForOsmNode(OsmNode node, OsmEntity way) {
         }
       }
 
-      if (node.isBarrier()) {
-        BarrierVertex bv = vertexFactory.barrier(nid, coordinate);
-        bv.setBarrierPermissions(node.overridePermissions(BarrierVertex.defaultBarrierPermissions));
-        iv = bv;
-      }
-
       if (includeOsmSubwayEntrances && node.isSubwayEntrance()) {
         String ref = node.getTag("ref");
-        iv = vertexFactory.stationEntrance(nid, coordinate, ref, node.wheelchairAccessibility());
+        iv = vertexFactory.stationEntrance(
+          nid,
+          coordinate,
+          ref,
+          node.explicitWheelchairAccessibility()
+        );
+      }
+
+      if (iv == null && (node.isBarrier() || isNodeOnLinearBarrier)) {
+        iv = vertexFactory.barrier(nid, coordinate, node.explicitWheelchairAccessibility());
+      }
+
+      if (iv instanceof BarrierVertex bv) {
```

Comment:
It seems that these 2 blocks can  be merged, right? iv is known to b e a barrierVertex here.

Reply:
iv can be a BoardingLocation

Reply:
I am also changing StationEntranceVertex to a subclass of BarrierVertex as well so they can't be merged.